### PR TITLE
Keep certain function that are potentially used in the debugger

### PR DESCRIPTION
**Pull request by augusto2112**
Currently, when compiling with no optimizations on, we still delete
functions that are sometimes used in the debugger. For example, users
might want to call functions which are unused, or compiler generated
setters/getters.

rdar://101046198

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

File: lib/SIL/IR/SILFunction.cpp [hunk 1]
```diff
@@ -872,6 +875,42 @@ SILFunction::isPossiblyUsedExternally() const {
   return swift::isPossiblyUsedExternally(linkage, getModule().isWholeModule());
 }
 
+bool SILFunction::shouldBePreservedForDebugger() const {
+  /* if (isGlobalInit()) */
```
Comment (nate-chandler) [inline lib/SIL/IR/SILFunction.cpp:879]: Should this comment be deleted?

**Review by nate-chandler — APPROVED** [review]
LGTM

**Comment by augusto2112** [comment]
@swift-ci smoke test

**Comment by eeckstein** [comment]
@swift-ci benchmark

**Pull request merged**